strawberry/printer/ast_from_value.py

```diff
@@ -0,0 +1,145 @@
+import dataclasses
+import re
+from math import isfinite
+from typing import Any, Mapping, Optional, cast
+
+from graphql.language import (
+    BooleanValueNode,
+    EnumValueNode,
+    FloatValueNode,
+    IntValueNode,
+    ListValueNode,
+    NameNode,
+    NullValueNode,
+    ObjectFieldNode,
+    ObjectValueNode,
+    StringValueNode,
+    ValueNode,
+)
+from graphql.pyutils import Undefined, inspect, is_iterable
+from graphql.type import (
+    GraphQLID,
+    GraphQLInputObjectType,
+    GraphQLInputType,
+    GraphQLList,
+    GraphQLNonNull,
+    is_enum_type,
+    is_input_object_type,
+    is_leaf_type,
+    is_list_type,
+    is_non_null_type,
+)
+
+
+__all__ = ["ast_from_value"]
+
+_re_integer_string = re.compile("^-?(?:0|[1-9][0-9]*)$")
+
+
+def ast_from_leaf_type(
+    serialized: object, type_: Optional[GraphQLInputType]
+) -> ValueNode:
+    # Others serialize based on their corresponding Python scalar types.
+    if isinstance(serialized, bool):
+        return BooleanValueNode(value=serialized)
+
+    # Python ints and floats correspond nicely to Int and Float values.
+    if isinstance(serialized, int):
+        return IntValueNode(value=str(serialized))
+    if isinstance(serialized, float) and isfinite(serialized):
+        value = str(serialized)
+        if value.endswith(".0"):
+            value = value[:-2]
```

Comment:
why do you single out the `.0` case? Is there a common scenario that leads to this where you'd want it printed as an integer? I notice you don't have a test-case for this either

Reply:
the test is here: https://github.com/strawberry-graphql/strawberry/blob/63f17292e18eadde6d30c71c0174369b665fa776/tests/test_printer/test_basic.py#L138

This was done here: https://github.com/graphql-python/graphql-core/pull/164

I think we should/could refine the logic at some point, but I also don't want to change the current printing behaviour 😊